source/client-side-encryption/client-side-encryption.rst

```diff
@@ -242,70 +334,318 @@ MongoClient Changes
       private Optional<MongoClient> internal_client; // An internal MongoClient. Created if no external keyVaultClient was set, or if a metadataClient is needed
    }
 
+.. glossary::
+
+   ``libmongocrypt_handle``
+
+      This is a handle to the `libmongocrypt` library and associated context.
+      There must be one per client that wishes to use `libmongocrypt`.
+
+   ``mongocryptd_client``
+
+      This is a regular ``MongoClient`` that talks to `mongocryptd` for
+      command marking. This client is not required if `csfle` is in use.
+
+AutoEncryptionOpts
+^^^^^^^^^^^^^^^^^^
+
+.. code:: typescript
+
    class AutoEncryptionOpts {
+      bypassAutoEncryption: Optional<Boolean>; // Default false.
       keyVaultClient: Optional<MongoClient>;
       keyVaultNamespace: String;
       kmsProviders: Map<String, Map<String, Value>>;
       schemaMap: Optional<Map<String, Document>>; // Maps namespace to a local schema
-      bypassAutoEncryption: Optional<Boolean>; // Default false.
       extraOptions: Optional<Map<String, Value>>;
       tlsOptions: Optional<Map<String, TLSOptions>>; // Maps KMS provider to TLS options.
    }
 
-A MongoClient can be configured to automatically encrypt collection
-commands and decrypt results.
+.. option:: bypassAutoEncryption
 
-Drivers MUST document that auto encryption is an enterprise-only
-feature and that auto encryption only occurs on collection level
-operations by including the following in the driver documentation for
-AutoEncryptionOpts:
+   :type: :ts:`boolean | undefined`
 
-   Automatic encryption is an enterprise only feature that only applies to
-   operations on a collection. Automatic encryption is not supported for
-   operations on a database or view, and operations that are not bypassed
-   will result in error (see `libmongocrypt: Auto Encryption Allow-List`_).
-   To bypass automatic encryption for all operations, set
-   bypassAutoEncryption=true in AutoEncryptionOpts.
+   Drivers MUST disable auto encryption when the 'bypassAutoEncryption' option
+   is |true| and not try to
+   `spawn mongocryptd <cse.managing-mongocryptd>` nor
+   `load csfle <cse.enabling-csfle>`. Automatic encryption may be
```

Comment:
Is it an explicit error to specify CSFLE library path options and `bypassAutoEncryption`, or will the CSFLE library path just be ignored? Either way is good for me.